kratos/includes/key_hash.h

```diff
@@ -219,6 +219,84 @@ namespace Kratos
         }
     };
 
+    /**
+     * @brief This is a hasher for indexed objects
+     * @tparam TIndexedObject The type of indexed object to be hashed
+     */
+    template<class TIndexedObject>
+    struct IndexedObjectHasher
+    {
+        /**
+         * @brief This is the () operator
+         * @param rIndexedObject The indexed object to be hashed
+         * @return The corresponding hash
+         */
+        HashType operator()(const TIndexedObject& rIndexedObject) const
+        {
+            return rIndexedObject.Id();
+        }
+    };
+
+    /**
+     * @brief This is a key comparer between two indexed objects
+     * @tparam TIndexedObject The type of indexed object to be compared
+     */
+    template<class TIndexedObject>
```

Comment:
why does it have to be a template if it is intended for the IndexedObject?

Reply:
i did not catch this...but indeed it should work without template for objects derived from IndexedObject